pkg/client/selfhosted/selfhosted.go

```diff
@@ -270,3 +277,24 @@ func (c *Client) setupBasicAuth(ctx context.Context, url string) (string, error)
 
 	return response.Token, nil
 }
+
+func newTLSConfig(insecure bool, CAPath string) (*tls.Config, error) {
+	// Load system CA Certs and/or create a new CertPool
+	rootCAs, _ := x509.SystemCertPool()
+	if rootCAs == nil {
+		rootCAs = x509.NewCertPool()
+	}
+
+	if CAPath != "" {
+		certs, err := ioutil.ReadFile(CAPath)
```

Comment:
The only thing that springs to mind here is how this RootCA could be rotated. Maybe this is "fine" for now? we might want to consider https://github.com/dyson/certman to ensure the tls config changes when the file does.